ibis/expr/rewrites.py

```diff
@@ -278,21 +282,39 @@ def rewrite_filter_input(value):
 
 
 @replace(p.Analytic | p.Reduction)
-def window_wrap_reduction(_, window):
-    # Wrap analytic and reduction functions in a window function. Used in the
-    # value.over() API.
+def window_wrap_reduction(_: ops.Analytic | ops.Reduction, window: LegacyWindowBuilder):
+    # Wrap analytic and reduction functions in a window function, merging their ordering.
+    # Used in the value.over() API.
+    order_by_specified_in_op = ("order_by" in _.argnames) and bool(_.order_by)
```

Comment:
This was an attempt to future proof us for when we add some aggregation in the future that has ordering. I want this logic to also apply to that new op, because I KNOW that we will forget to add that new op to the test I'm adding in this PR ;).

I'm relying on the assumption that the new op will have an argument called order_by. This is slightly brittle, if we made it so all ordered aggregation classes explicitly subclassed from some Ordered ABC then this check could be a little more official.